src/librustc_mir/transform/generator.rs

```diff
@@ -332,9 +332,39 @@ impl<'tcx> Visitor<'tcx> for StorageIgnored {
     }
 }
 
+struct BorrowedLocals(liveness::LocalSet);
+
+fn mark_as_borrowed<'tcx>(lvalue: &Lvalue<'tcx>, locals: &mut BorrowedLocals) {
+    match *lvalue {
+        Lvalue::Local(l) => { locals.0.add(&l); },
+        Lvalue::Static(..) => (),
+        Lvalue::Projection(ref proj) => {
+            match proj.elem {
+                // For derefs there's no underlying local that gets borrowed
+                // It is either a Box or some already borrowed local.
```

Comment:
```Rust
// It is either a Box or some already borrowed local.
```

nit: it can also be a borrow from something based on an upvar, e.g. `&fcx.tcx`.

Reply:
I was thinking of MIR locals here.

Reply:
`(*fcx).tcx`  is not an already borrowed MIR local, it's from the place where `fcx` was allocated from, which can be everywhere.

Reply:
I changed this comment.